circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala

```diff
@@ -78,6 +135,30 @@ abstract class CirceAkkaSerializer[Ser <: AnyRef: ClassTag](system: ExtendedActo
     }
   }
 
+  /**
+   * The intended usage of this method is to provide any form of support for generic classes.
+   *
+   * Because of type erasure, it's impossible to [[org.virtuslab.ash.circe.Register]] one generic class two times with different type parameters.
+   * The trick for combating type erasure is to register generic class only once with type parameter being its upper bound, and provide custom made [[io.circe.Codec]] that can serialize/deserialize all classes that are used as a type parameter.
+   *
+   * For example, if the upper bound is `Any`, but you know that only `Int` and `String` are used as a type parameter, then you can create a custom [[io.circe.Codec]] for `Any` that handles `Int` and `String` and throws `Exception` otherwise.
+   *
+   * To use this method correctly, set the upper bound for the type parameter of generic class to `Ser` and put the returned Codec as implicit in a place that can be seen by type derivation.
+   *
+   * Example of generic class:
+   * {{{
+   *   case class GenericClass[A <: MySerializable, B <: MySerializable](a: A, b: B) extends MySerializable
+   * }}}
+   * and its registration in serializer:
+   * {{{
+   *   Register[GenericClass[MySerializable, MySerializable]]
+   * }}}
+   *
+   * @return [[io.circe.Codec]] that can serialize all subtypes of `Ser`
+   */
+  protected def genericCodec: Codec[Ser] = Codec.from(genericDecoder, genericEncoder)
+
+  //TODO make private
```

Comment:
Whoops while you're here... stray TODOs without issue number are a ♋ and should be banned

Pls add a script like https://github.com/VirtusLab/git-machete-intellij-plugin/blob/develop/scripts/enforce-issue-number-for-todos and call it from CI